docs/experiments/extract-mode.md

```diff
@@ -89,7 +89,7 @@ The extract mode specific fields are further explained in two sections below.
 
 ### `extract`
 
-The `extract` key is required when in extract mode. The value must be a metavariable appearing in your pattern(s), and the value bound to that metavariable will be extracted.
+The `extract` key is required when in extract mode. The value must be a metavariable appearing in your pattern(s), and Semgrep extracts the value bound to that metavariable.
```

Comment:
> Use active voice - is extracted - Semgrep extracts - what do we mean by extracting it actually? Detected maybe?


I wasn't sure how much to talk about the actual mechanics of what we do for a page like this, but the "extraction" essentially comprises the process of reading the text bound to the metavariable from the original file, writing it to a temporary file, and then running rules for the destination language against that file.

It is unclear to me if that description is more or less helpful than the admittedly nebulous "extracts".

The suggestion you have seems good to me wrt your other points.

Reply:
Interesting mechanics @kopecs! Surely, I would avoid explaining it to external users as it is not needed for the use of it. So is the following suggestion good enough?

```suggestion
The `extract` key is required when in extract mode. The value must be a metavariable appearing in your pattern(s), and Semgrep detects the value bound to that metavariable.
```

Reply:
I would be concerned that "detects" is less clear. I think we could just be more explicit, without going into the level of detail again. Perhaps something like:

The `extract` key is required when in extract mode. The value must be a metavariable appearing in your pattern(s). The code bound to the metavariable is used for subsequent queries of non-extract mode rules targeting `dest-language`.

Reply:
I agree! Only avoid passive sentence to express the agent, so I would suggest:

```suggestion
The `extract` key is required when in extract mode. The value must be a metavariable appearing in your pattern(s). Semgrep uses the code bound to the metavariable for subsequent queries of non-extract mode rules targeting `dest-language`.
```